extensions/pkg/controller/extension/reconciler.go

```diff
@@ -173,39 +174,43 @@ func (r *reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 		err    error
 	)
 
-	if ex.DeletionTimestamp != nil {
-		return r.delete(r.ctx, ex)
-	}
+	operationType := gardencorev1beta1helper.ComputeOperationType(ex.ObjectMeta, ex.Status.LastOperation)
 
-	result, err = r.reconcile(r.ctx, ex)
-	if err != nil {
-		return result, err
+	switch {
+	case extensionscontroller.IsMigrated(ex):
+		return reconcile.Result{}, nil
+	case operationType == gardencorev1beta1.LastOperationTypeMigrate:
+		return r.migrate(r.ctx, ex)
+	case ex.DeletionTimestamp != nil:
+		return r.delete(r.ctx, ex)
+	case ex.Annotations[v1beta1constants.GardenerOperation] == v1beta1constants.GardenerOperationRestore:
+		if result, err = r.restore(r.ctx, ex, operationType); err != nil {
+			return result, err
+		}
+		return reconcile.Result{Requeue: r.resync != 0, RequeueAfter: r.resync}, nil
+	default:
+		if result, err = r.reconcile(r.ctx, ex, operationType); err != nil {
+			return result, err
+		}
+		return reconcile.Result{Requeue: r.resync != 0, RequeueAfter: r.resync}, nil
```

Comment:
Hm, why do we have this special handing with the error here? I don't see this for the infrastructure controller, for example: https://github.com/gardener/gardener/blob/master/extensions/pkg/controller/infrastructure/reconciler.go#L104-L115

Can you explain?

Reply:
Monkey see monkey do.

Reply:
Hmm I guess we should keep it then?

Reply:
cc @timuthy - can you give an opinion? I think you were contributing it initially (hope I remember correctly)

Reply:
Are you specifically asking for the re-queuing? The motivation is that an extension can make use of an automatic re-queuing to allow clean-up jobs, for example. But I don't see a necessity for doing this in the restore case.

Reply:
OK, thanks. Yes, agreed, for restore it's probably not needed. @vlvasilev can you adapt accordingly?

Reply:
Done